### PR TITLE
Minor updates to OpenTelemetry sample

### PR DESCRIPTION
**Pull request by trask**
Thanks for this excellent OpenTelemetry sample, it was easy to get working!

A couple of minor updates I wanted to contribute back if they make sense.

### CODE REVIEW

**Comment by trask** [comment]
cc: @samvaity

File: sdk/core/azure-core-tracing-opentelemetry/src/samples/AsyncListKeyVaultSecretsSample.md [hunk 1]
```diff
@@ -10,12 +10,17 @@ Sample uses **[opentelemetry-sdk][opentelemetry_sdk]** as implementation package
...
     </dependency>
     <dependency>
         <groupId>com.azure</groupId>
         <artifactId>azure-security-keyvault-secrets</artifactId>
-        <version>4.0.0</version>
+        <version>4.1.0-beta.1</version>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
```
Comment (samvaity) [inline sdk/core/azure-core-tracing-opentelemetry/src/samples/AsyncListKeyVaultSecretsSample.md]: @g2vinay Can you confirm if this dependency needs to be explicitly added for using KV secrets package?

Comment (g2vinay) [reply sdk/core/azure-core-tracing-opentelemetry/src/samples/AsyncListKeyVaultSecretsSample.md]: Yeah, KV doesn't take compile dependency on Identity.
All the track 2 SDKs, should need to have this explicitly, if they're using Identity on auth.

**Comment by joshfree** [comment]
@samvaity I believe this open PR has been superseded by more recent OT sample PR?  Can this be closed?

**Comment by samvaity** [comment]
Closing as handled in #10846

**Pull request closed without merging**